provider/docker/docker.go

```diff
@@ -153,15 +153,18 @@ func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *s
 			if p.Watch {
 				ctx, cancel := context.WithCancel(ctx)
 				if p.SwarmMode {
+					errChan := make(chan error)
 					// TODO: This need to be change. Linked to Swarm events docker/docker#23827
 					ticker := time.NewTicker(SwarmDefaultWatchTime)
 					pool.Go(func(stop chan bool) {
+						defer close(errChan)
```

Comment:
The receiver at line 185 blocks the execution until the channel is closed or received an error.